File: Source/Extensions/Tessa.Extensions.Default.Client/Views/FilterViewDialogDescriptors.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Tessa.Views.Metadata;
using Tessa.Views.Metadata.Criteria;

namespace Tessa.Extensions.Default.Client.Views
{
    /// <summary>
    /// Предоставляет объекты типа <see cref="FilterViewDialogDescriptor"/>.
    /// </summary>
    public static class FilterViewDialogDescriptors
    {
        #region Constants And Static Fields

        /// <summary>
        /// Дескриптор, описывающий специальный диалог с параметрами фильтрации представления РМ Администратор/Тестирование/Автомобили.
        /// </summary>
        public static readonly FilterViewDialogDescriptor Cars =
            new FilterViewDialogDescriptor(
                "CarViewParameters",
                new ReadOnlyCollection<ParameterMapping>(
                    new List<ParameterMapping>()
                    {
                        new ParameterMapping(
                            "CarName",
                            "Parameters",
                            "Name"),
                        new ParameterMapping(
                            "CarMaxSpeed",
                            "Parameters",
                            "MaxSpeed")
                        {
                            CriteriaOperator = CriteriaHelper.GetCriteria(CriteriaOperatorConst.Equality),
                        },
                        new ParameterMapping(
                            "Driver",
                            "Parameters",
                            "DriverID")
                        {
                            DisplayValueSectionName = "Parameters",
                            DisplayValueFieldName = "DriverName",
                        },
                        new ParameterMapping(
                            "CarReleaseDateFrom",
                            "Parameters",
                            "ReleaseDateFrom"),
                        new ParameterMapping(
                            "CarReleaseDateTo",
                            "Parameters",
                            "ReleaseDateTo"),
                    }));

        #endregion
    }
}

File: WebClient SDK/src/default/views/filterViewDialogDescriptors.ts
import { CriteriaOperatorConst, getCriteria } from 'tessa/views/metadata';

import { FilterViewDialogDescriptor } from './filterViewDialogDescriptor';

/**
 * Предоставляет объекты типа {@link FilterViewDialogDescriptor}.
 */
export class FilterViewDialogDescriptors {
  /**
   * Дескриптор, описывающий специальный диалог с параметрами фильтрации представления РМ Администратор/Тестирование/Автомобили.
   */
  static readonly cars: FilterViewDialogDescriptor = {
    dialogName: 'CarViewParameters',
    parametersMapping: [
      { alias: 'CarName', valueSectionName: 'Parameters', valueFieldName: 'Name' },
      {
        alias: 'CarMaxSpeed',
        valueSectionName: 'Parameters',
        valueFieldName: 'MaxSpeed',
        criteriaOperator: getCriteria(CriteriaOperatorConst.Equality)
      },
      {
        alias: 'Driver',
        valueSectionName: 'Parameters',
        valueFieldName: 'DriverID',
        displayValueSectionName: 'Parameters',
        displayValueFieldName: 'DriverName'
      },
      {
        alias: 'CarReleaseDateFrom',
        valueSectionName: 'Parameters',
        valueFieldName: 'ReleaseDateFrom'
      },
      { alias: 'CarReleaseDateTo', valueSectionName: 'Parameters', valueFieldName: 'ReleaseDateTo' }
    ]
  };
}
